lib/processors/manifestTransformer.js

```diff
@@ -0,0 +1,218 @@
+import semver from "semver";
+const {SemVer: Version} = semver;
+import {getLogger} from "@ui5/logger";
+const log = getLogger("builder:processors:manifestTransformer");
+
+const APP_DESCRIPTOR_V22 = new Version("1.21.0");
+
+/**
+ * Transforms i18n_en.properties to en
+ *
+ * @param {string} name
+ */
+function getLocale(name) {
+	return name.substring(name.lastIndexOf("_") + 1).replaceAll(".properties", "");
+}
+
+/**
+ * Determines if manifest contain handlebars placeholder templates
+ *
+ * Only in case of placeholders in manifest.json the bundle configured
+ * in sap.app/i18n is requested by sap/ui/core/Manifest.js
+ *
+ * @param {object} manifest The manifest
+ * @returns {boolean}
+ */
+function hasManifestTemplates(manifest) {
+	const rManifestTemplate = /\{\{([^}}]+)\}\}/g;
+	return JSON.stringify(manifest).match(rManifestTemplate);
+}
+
+function isAutofillRequired(bundleConfig, appId) {
+	const {supportedLocales, bundleUrl, bundleName} = bundleConfig;
+
+	// TODO: check is not correct, if the manifest.json is in a subfolder the bundleUr
+	// might start with a "..", see how to know bundle is outside the project
+	if (bundleUrl && (bundleUrl.startsWith("..") || bundleUrl.startsWith("/"))) {
+		return false;
+	}
+
+	// TODO: bundle is not inside the manifest related project but still part of UI5 project
+	if (bundleName && !bundleName.startsWith(appId)) {
+		return false;
+	}
+
+	// required locales are defined, so do nothing
+	if (supportedLocales) {
+		return false;
+	}
+
+	return true;
+}
+
+function getType(manifest) {
+	return manifest["sap.app"].type;
+}
+
+function isFallbackLocalePartOfSupportedLocales(fallbackLocale, supportedLocales) {
+	return supportedLocales.indexOf(fallbackLocale) > -1;
+}
+
+function getSapAppBundle(manifest, bundleConfigs) {
+	const sapAppConfig = manifest["sap.app"];
+	let sapAppBundleConfig = sapAppConfig?.i18n;
+
+	if (getType(manifest) !== "library") {
+		if (hasManifestTemplates(manifest)) {
+			// sap/ui/core/Manifest defaults sap.app/i18n to "i18n/i18n.properties"
+			// when template is used in manifest.json
+			sapAppBundleConfig = sapAppBundleConfig || "i18n/i18n.properties";
```

Comment:
yes bundle is only loaded when the manifest.json contains a template e.g `{{TITLE}}` and this is checked by in the processor too

Reply:
I don't feel like we should couple that default to the presence of a template. The default always applies. The runtime just doesn't do the "defaulting" if there's no template, as it doesn't need to load the ResourceBundle. But this might change in future. The runtime might in future decide to load the ResourceBundle for other purposes. So in the tooling, we should always consider this default, and add the `supportedLocales`.